File: src/main/java/juuxel/woodsandmires/block/PineSaplingGenerator.java
package juuxel.woodsandmires.block;

import juuxel.woodsandmires.feature.WamConfiguredFeatureKeys;
import net.minecraft.block.sapling.SaplingGenerator;
import net.minecraft.registry.RegistryKey;
import net.minecraft.util.math.random.Random;
import net.minecraft.world.gen.feature.ConfiguredFeature;

public class PineSaplingGenerator extends SaplingGenerator {
    @Override
    protected RegistryKey<ConfiguredFeature<?, ?>> getTreeFeature(Random random, boolean bees) {
        return WamConfiguredFeatureKeys.PINE_FROM_SAPLING;
    }
}

File: src/main/kotlin/juuxel/woodsandmires/block/PineSaplingGenerator.kt
package juuxel.woodsandmires.block

import java.util.Random
import juuxel.woodsandmires.feature.WamConfiguredFeatures
import net.minecraft.block.sapling.SaplingGenerator
import net.minecraft.world.gen.feature.ConfiguredFeature
import net.minecraft.world.gen.feature.TreeFeatureConfig

object PineSaplingGenerator : SaplingGenerator() {
    override fun createTreeFeature(random: Random, flowerRelated: Boolean): ConfiguredFeature<TreeFeatureConfig, *> =
        WamConfiguredFeatures.PINE_FROM_SAPLING
}
